src/textual/app.py

```diff
@@ -1026,6 +1069,54 @@ async def _press_keys(self, keys: Iterable[str]) -> None:
                 await app._animator.wait_until_complete()
                 await wait_for_idle(0)
 
+    def _flush(self, stderr: bool = False) -> None:
+        """Called when stdout or stderr is flushed.
+
+        Args:
+            stderr: True if the print was to stderr, or False for stdout.
+
+        """
+        if self._devtools_redirector is not None:
+            self._devtools_redirector.flush()
+
+    def _print(self, text: str, stderr: bool = False) -> None:
+        """Called with capture print.
+
+        Args:
+            text: Text that has been printed.
+            stderr: True if the print was to stderr, or False for stdout.
+        """
+        if self._devtools_redirector is not None:
+            self._devtools_redirector.write(text)
+        for target, (_stdout, _stderr) in self._capture_print.items():
+            if (_stderr and stderr) or (_stdout and not stderr):
+                target.post_message(events.Print(text, stderr=stderr))
+
+    def capture_print(
```

Comment:
More to note than anything else, but I feel this might be better as `begin_capture_print`; it mirrors `end_capture_print`, makes the intent more clear, and then later allows for `capture_print` to be used as a context.

Reply:
Yeah, although there is a odd inconsistency then. There is a existing `capture_mouse` which isn't a context manager.